packages/terra-data-grid/src/FlowsheetDataGrid.jsx

```diff
@@ -157,6 +157,38 @@ function FlowsheetDataGrid(props) {
     isResizable: false,
   })), [columns]);
 
+  const displayedColumnsWithColumnSpan = useMemo(() => {
+    let i = 0;
+    const newDisplayedColumns = [];
+    columns.forEach((column) => {
+      if (column.columnSpan > 1) {
+        newDisplayedColumns[i] = {
+          ...column,
+          columnSpanIndex: 0,
+          isSelectable: false,
+          isSelected: false,
+          isResizable: false,
+        };
+        i += 1;
+        for (let counter = column.columnSpan; counter > 1; counter -= 1) {
+          newDisplayedColumns[i] = {
+            id: `${column.id}`, columnSpanIndex: `${column.columnSpan - counter + 1}`, isSelectable: false, isResizable: false, isSelected: false,
+          };
+          i += 1;
+        }
+      } else {
+        newDisplayedColumns[i] = {
+          ...column,
+          isSelectable: false,
+          isSelected: false,
+          isResizable: false,
+        };
+        i += 1;
+      }
+    });
+    return newDisplayedColumns;
+  }, [columns]);
```

Comment:
It seems like we are duplicating this logic in the DataGrid and Flowsheet components.  Can it not be centralized to the DataGrid component?